R/categorize_prism.R

```diff
@@ -0,0 +1,75 @@
+#' Get PRISM activity level cutpoints for given
+#' diseases and locations
+#'
+#' @param diseases disease(s) for which to return the cutpoints.
+#' One of `"ARI"`, `"COVID-19"`, `"Influenza"`, or `"RSV"`, or
+#' an array of those values.
+#' @param locations location(s) for which to return the cutpoints.
+#' A location two-letter abbreviation as in the `short_name`
+#' column of [forecasttools::us_location_table], or an array
+#' of those abbreviations.
+#' @return The cutpoints, as an ordered list of vectors.
+#'
+#' @examples
+#' get_prism_cutpoints("WA", "Influenza")
+#'
+#' get_prism_cutpoints(c("US", "WA"), "COVID-19")
+#'
+#' get_prism_cutpoints(c("US", "WA"), c("ARI", "RSV"))
+#' @export
+get_prism_cutpoints <- function(locations, diseases) {
+  checkmate::assert_names(
+    diseases,
+    subset.of =
+      dimnames(forecasttools::prism_thresholds)$disease,
+    what = "disease"
+  )
+  checkmate::assert_names(
+    locations,
+    subset.of =
+      dimnames(forecasttools::prism_thresholds)$location,
+    what = "location"
+  )
+
+  return(purrr::map2(locations, diseases, \(x, y) {
+    forecasttools::prism_thresholds[x, y, ]
+  }))
+}
```

Comment:
This feels inefficient but equally it's readable and premature optimization is bad

Reply:
I can't find any way to avoid it because how R interprets array slicing.